src/progress-indicator/progress-indicator.component.ts

```diff
@@ -13,21 +13,36 @@ import { ExperimentalService } from "./../experimental.module";
 			'bx--progress--vertical': (orientation === 'vertical')
 		}">
 		<li
-		class="bx--progress-step bx--progress-step--{{step.state}}"
-		*ngFor="let step of steps; let i = index">
-			<ibm-icon-checkmark-outline16 *ngIf="step.state == 'complete'"></ibm-icon-checkmark-outline16>
-			<svg *ngIf="step.state == 'current'">
-				<path *ngIf="isExperimental" d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0" ></path>
-			</svg>
-			<svg *ngIf="step.state == 'incomplete'">
-				<path
-					*ngIf="isExperimental"
-					d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z">
-				</path>
-			</svg>
-			<ibm-icon-error-outline16 *ngIf="step.state == 'error'"></ibm-icon-error-outline16>
-			<p class="bx--progress-label">{{step.text}}</p>
-			<span class="bx--progress-line"></span>
+			class="bx--progress-step bx--progress-step--{{step.state[0]}}"
+			*ngFor="let step of steps; let i = index"
+			[ngClass]="{'bx--progress-step--disabled' : step.disabled}">
+			<div class="bx--progress-step-button bx--progress-step-button--unclickable" role="button" tabindex="-1">
+				<ibm-icon-checkmark-outline16 *ngIf="step.state == 'complete'"></ibm-icon-checkmark-outline16>
+				<svg *ngIf="step.state == 'current'">
+					<path *ngIf="isExperimental" d="M 7, 7 m -7, 0 a 7,7 0 1,0 14,0 a 7,7 0 1,0 -14,0" ></path>
+				</svg>
+				<svg *ngIf="step.state == 'incomplete'">
+					<path
+						*ngIf="isExperimental"
+						d="M8 1C4.1 1 1 4.1 1 8s3.1 7 7 7 7-3.1 7-7-3.1-7-7-7zm0 13c-3.3 0-6-2.7-6-6s2.7-6 6-6 6 2.7 6 6-2.7 6-6 6z">
+					</path>
+				</svg>
+				<svg *ngIf="step.state.includes('error')" width="16" height="16" viewBox="0 0 16 16" class="bx--progress__warning">
```

Comment:
Is there an icon for this?

Reply:
If I use the icon it won't be able to apply the `bx--progress__warning` class to it

Reply:
What does the selector look like? You may be able to use `innerClass` ... this applies to all cases here